### PR TITLE
Fast indexing of technique variants in decoder.

### PR DESCRIPTION
**Pull request by zbigg**
Fast indexing of technique variant in decoder.
    
Identify common technique handlers only searching through (usually)
short list of technique variable parameters that can be compared very
efficiently.

Related-to: HARP-8858

### CODE REVIEW

File: @here/harp-datasource-protocol/lib/ExprPool.ts [hunk 1]
```diff
@@ -33,17 +42,24 @@ export class ExprPool implements ExprVisitor<Expr, void> {
...
+    private readonly m_caseExprs: CaseExpr[] = [];
     private readonly m_callExprs = new Map<string, CallExpr[]>();
 
+    private nextKey: number = 1;
     /**
      * Add `expr` to this [[ExprPool]] and return a unique [[Expr]]
      * object that is structurally equivalent to `expr`.
      *
-     * @param expr The [[Expr]] to add to this [[ExprPool]].
+     * @param expr The [[PooledExpr]] to add to this [[ExprPool]].
```
Comment (ninok) [inline @here/harp-datasource-protocol/lib/ExprPool.ts]: Minor typo: `@param` is an `Expr` and `@return` is a `PooledExpr`

**Comment by zbigg** [comment]
@heremaps-bot retest this please

File: @here/harp-datasource-protocol/lib/ExprPool.ts (hunk 1 above)
Comment (zbigg) [reply @here/harp-datasource-protocol/lib/ExprPool.ts]: Fixed, thanks.

**Comment by zbigg** [comment]
Decoders tests please

**Comment by zbigg** [comment]
Decoders tests please

File: @here/harp-datasource-protocol/test/StyleSetEvaluatorTest.ts [hunk 2]
```diff
@@ -176,46 +176,6 @@ describe("StyleSetEvaluator", function() {
             assert.equal((r2[0] as SolidLineTechnique).lineWidth, 3);
             assert.equal(r2[0].renderOrder, 3);
         });
-
```
Comment (zbigg) [inline @here/harp-datasource-protocol/test/StyleSetEvaluatorTest.ts]: This test was obsoleted by #1338

**Pull request closed without merging**